{
    "url": "https://api.github.com/repos/spack/spack/issues/24770",
    "repository_url": "https://api.github.com/repos/spack/spack",
    "labels_url": "https://api.github.com/repos/spack/spack/issues/24770/labels{/name}",
    "comments_url": "https://api.github.com/repos/spack/spack/issues/24770/comments",
    "events_url": "https://api.github.com/repos/spack/spack/issues/24770/events",
    "html_url": "https://github.com/spack/spack/issues/24770",
    "id": 939556261,
    "node_id": "MDU6SXNzdWU5Mzk1NTYyNjE=",
    "number": 24770,
    "title": "External Python run outside build environment",
    "user": {
        "login": "billyauhk",
        "id": 3797567,
        "node_id": "MDQ6VXNlcjM3OTc1Njc=",
        "avatar_url": "https://avatars.githubusercontent.com/u/3797567?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/billyauhk",
        "html_url": "https://github.com/billyauhk",
        "followers_url": "https://api.github.com/users/billyauhk/followers",
        "following_url": "https://api.github.com/users/billyauhk/following{/other_user}",
        "gists_url": "https://api.github.com/users/billyauhk/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/billyauhk/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/billyauhk/subscriptions",
        "organizations_url": "https://api.github.com/users/billyauhk/orgs",
        "repos_url": "https://api.github.com/users/billyauhk/repos",
        "events_url": "https://api.github.com/users/billyauhk/events{/privacy}",
        "received_events_url": "https://api.github.com/users/billyauhk/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [
        {
            "id": 73908754,
            "node_id": "MDU6TGFiZWw3MzkwODc1NA==",
            "url": "https://api.github.com/repos/spack/spack/labels/bug",
            "name": "bug",
            "color": "fc2929",
            "default": true,
            "description": null
        },
        {
            "id": 1433532775,
            "node_id": "MDU6TGFiZWwxNDMzNTMyNzc1",
            "url": "https://api.github.com/repos/spack/spack/labels/triage",
            "name": "triage",
            "color": "ed9793",
            "default": false,
            "description": "The issue needs to be prioritized"
        }
    ],
    "state": "open",
    "locked": false,
    "assignee": {
        "login": "adamjstewart",
        "id": 12021217,
        "node_id": "MDQ6VXNlcjEyMDIxMjE3",
        "avatar_url": "https://avatars.githubusercontent.com/u/12021217?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/adamjstewart",
        "html_url": "https://github.com/adamjstewart",
        "followers_url": "https://api.github.com/users/adamjstewart/followers",
        "following_url": "https://api.github.com/users/adamjstewart/following{/other_user}",
        "gists_url": "https://api.github.com/users/adamjstewart/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/adamjstewart/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/adamjstewart/subscriptions",
        "organizations_url": "https://api.github.com/users/adamjstewart/orgs",
        "repos_url": "https://api.github.com/users/adamjstewart/repos",
        "events_url": "https://api.github.com/users/adamjstewart/events{/privacy}",
        "received_events_url": "https://api.github.com/users/adamjstewart/received_events",
        "type": "User",
        "site_admin": false
    },
    "assignees": [
        {
            "login": "adamjstewart",
            "id": 12021217,
            "node_id": "MDQ6VXNlcjEyMDIxMjE3",
            "avatar_url": "https://avatars.githubusercontent.com/u/12021217?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/adamjstewart",
            "html_url": "https://github.com/adamjstewart",
            "followers_url": "https://api.github.com/users/adamjstewart/followers",
            "following_url": "https://api.github.com/users/adamjstewart/following{/other_user}",
            "gists_url": "https://api.github.com/users/adamjstewart/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/adamjstewart/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/adamjstewart/subscriptions",
            "organizations_url": "https://api.github.com/users/adamjstewart/orgs",
            "repos_url": "https://api.github.com/users/adamjstewart/repos",
            "events_url": "https://api.github.com/users/adamjstewart/events{/privacy}",
            "received_events_url": "https://api.github.com/users/adamjstewart/received_events",
            "type": "User",
            "site_admin": false
        }
    ],
    "milestone": null,
    "comments": 3,
    "created_at": "2021-07-08T07:24:33Z",
    "updated_at": "2021-07-09T11:12:04Z",
    "closed_at": null,
    "author_association": "NONE",
    "active_lock_reason": null,
    "body": "We have existing lmod for Python, etc. and would like spack to leverage on them. However during installation of paraview, there are errors:\r\n\r\n### Steps to reproduce the issue\r\n\r\n```\r\n$ spack install paraview@5.9.1~cuda %gcc@10.2.0 examples=True target=skylake_avx512\r\n==> Warning: Missing a source id for libtiff@4.2.0\r\n==> Warning: Missing a source id for perl@5.26.3\r\n==> Warning: Missing a source id for ncurses@6.1.20180224\r\n[+] /usr (external bzip2-1.0.6-hqdangdfn3xwndx6dubjijwwrz7ukuuu)\r\n==> cmake@3.19.7 : has external module in ['cmake/3.19.7']\r\n[+] /share1/cmake/3.19.7 (external cmake-3.19.7-2ool2rf6ynzlgczlaikwytag7iau62hb)\r\n...\r\n==> Installing py-setuptools-50.3.2-zp6t4o7sx4euf2q5z33ypbg5rmtluruc\r\n==> No binary for py-setuptools-50.3.2-zp6t4o7sx4euf2q5z33ypbg5rmtluruc found: installing from source\r\n/share1/python/3.9.2/bin/python3.9: error while loading shared libraries: libpython3.9d.so.1.0: cannot open shared object file: No such file or directory\r\n/share1/python/3.9.2/bin/python3.9: error while loading shared libraries: libpython3.9d.so.1.0: cannot open shared object file: No such file or directory\r\n==> Error: ProcessError: Command exited with status 127:\r\n    '/share1/python/3.9.2/bin/python3.9' '-c' 'from distutils.sysconfig import get_config_var; print(get_config_var('\"'\"'CC'\"'\"'))'\r\n...\r\n==> Installing perl-data-dumper-2.173-aswdg37psj5xukn2m3bbq4zxvd4oalra\r\n==> No binary for perl-data-dumper-2.173-aswdg37psj5xukn2m3bbq4zxvd4oalra found: installing from source\r\n==> Warning: Missing a source id for perl@5.26.3\r\n==> Using cached archive: /share1/github/spack/var/spack/cache/_source-cache/archive/69/697608b39330988e519131be667ff47168aaaaf99f06bd2095d5b46ad05d76fa.tar.gz\r\n==> No patches needed for perl-data-dumper\r\n==> perl-data-dumper: Executing phase: 'configure'\r\n==> Error: ProcessError: Command exited with status 2:\r\n    '/usr/bin/perl' 'Makefile.PL' 'INSTALL_BASE=/share1/spack/centos8-skylake_avx512/gcc-10.2.0/perl-data-dumper-2.173-aswdg37psj5xukn2m3bbq4zxvd4oalra'\r\nSee build log for details:\r\n  /tmp/root/spack-stage/spack-stage-perl-data-dumper-2.173-aswdg37psj5xukn2m3bbq4zxvd4oalra/spack-build-out.txt\r\n...\r\n==> Error: paraview-5.9.1-jt32ulb26k2xaxwbmn2tpa2uvmkkvzpy: Package was not installed\r\n==> Error: Installation request failed.  Refer to reported errors for failing package(s).\r\n```\r\nProbably the error also affects Perl, but the Perl was an external using prefix (OS-installed package), not a lmod\r\n\r\n### Error Message\r\nBeing instructed on Slack to try this\r\n```\r\n$ spack build-env paraview@5.9.1~cuda %gcc@10.2.0 examples=True target=skylake_avx512 -- bash -c \"module list\"\r\n/share1/python/3.9.2/bin/python3.9: error while loading shared libraries: libpython3.9d.so.1.0: cannot open shared object file: No such file or directory\r\n/share1/python/3.9.2/bin/python3.9: error while loading shared libraries: libpython3.9d.so.1.0: cannot open shared object file: No such file or directory\r\n==> Error: Command exited with status 127:\r\n'/share1/python/3.9.2/bin/python3.9' '-c' 'from distutils.sysconfig import get_config_var; print(get_config_var('\"'\"'CC'\"'\"'))'\r\n```\r\n\r\n### Information on your system\r\n\r\n* **Spack:** 0.16.2-2976-e37d663881\r\n* **Python:** 3.6.8\r\n* **Platform:** linux-centos8-cascadelake\r\n* **Concretizer:** original\r\n\r\nRelevant section of Python at packages.yaml looks like this:\r\n```\r\npython:\r\n  externals:\r\n  - spec: python@3.9.2~bz2+ctypes+dbm+lzma~nis~pyexpat+readline~sqlite3+ssl~tix~tkinter+uuid+zlib\r\n    modules:\r\n    - python/3.9.2\r\n  - spec: python@3.6.8+bz2+ctypes+dbm+lzma+nis+pyexpat+readline+sqlite3+ssl~tix~tkinter+uuid+zlib\r\n    prefix: /usr\r\n```\r\n\r\n### Additional information\r\n\r\n<!-- These boxes can be checked by replacing [ ] with [x] or by clicking them after submitting the issue. -->\r\n- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform\r\n- [x] I have searched the issues of this repo and believe this is not a duplicate\r\n- [x] I have run the failing commands in debug mode and reported the output\r\n\r\nAttached outputs from:\r\n```spack install paraview@5.9.1~cuda %gcc@10.2.0 examples=True target=skylake_avx512 > spack_paraview_python_error.txt```\r\n[spack_paraview_python_error.txt](https://github.com/spack/spack/files/6782440/spack_paraview_python_error.txt)\r\n\r\n```spack -d build-env paraview@5.9.1~cuda %gcc@10.2.0 examples=True target=skylake_avx512 -- bash -c \"module list\" > spack_builenv.txt```\r\n[spack_buildenv.txt](https://github.com/spack/spack/files/6782434/spack_buildenv.txt)\r\n\r\n\r\n",
    "performed_via_github_app": null
}